README.md

```diff
@@ -0,0 +1,223 @@
+- <link integrity="" rel="stylesheet" href="{{rootURL}}assets/my-classic-app.css">
+- <link integrity="" rel="stylesheet" href="{{rootURL}}assets/test-support.css">
++ <link integrity="" rel="stylesheet" href="/@embroider/virtual/vendor.css">
++ <link integrity="" rel="stylesheet" href="/@embroider/virtual/app.css">
++ <link integrity="" rel="stylesheet" href="/@embroider/virtual/test-support.css">
+
+- <script src="{{rootURL}}assets/vendor.js"></script>
+- <script src="{{rootURL}}assets/test-support.js"></script>
++ <script src="/@embroider/virtual/vendor.js"></script>
++ <script src="/@embroider/virtual/test-support.js"></script>
+
+- <script src="{{rootURL}}assets/my-classic-app.js"></script>
+- <script src="{{rootURL}}assets/tests.js"></script>
++ <script type="module">
++  import { start } from './test-helper';
++   import.meta.glob("./**/*.{js,ts,gjs,gts}", { eager: true });
++   start();
++ </script>
+
+- {{content-for "test-body-footer"}}
+```
+
+### Running code replacements on... `app/config/environment.js`
+
+Since the file `app/config/environment.js` is created out of the app blueprint, it has a placeholder `<%= name %>` for your app name. Replace it with the correct value. The name to use can be read in `ENV.modulePrefix` in your `config/environment.js`.
+
+### Running code replacements on... `ember-cli-build.js`
+
+Instead of building the app directly with Broccoli, we use `@embroider/compat`, a module that essentially serves as a bridge between classic apps and modern Vite apps. The Broccoli app instance is still there, but it's passed in argument to Embroider.
+
+Considering an empty 6.2 Ember app, the codemod does the following:
+
+```diff
+ 'use strict';
+
+  const EmberApp = require('ember-cli/lib/broccoli/ember-app');
++ const { compatBuild } = require('@embroider/compat');
+
+  module.exports = function (defaults) {
++   const { buildOnce } = await import('@embroider/vite');
+    const app = new EmberApp(defaults, {
+      // Add options here
+    });
+
+-   return app.toTree();
++   return compatBuild(app, buildOnce);
+  };
+```
+
+### Running code replacements on... `testem.js`
+
+The change in this file introduces a new way to prevent its execution when tests run directly in the browser.
+
+The codemod looks for the `module.exports` and wraps it in a conditional that checks the `module` existence.
+
+```diff
+  'use strict';
+
++ if (typeof module !== 'undefined') {
+    module.exports = { ... }
++ }
+```
+
+### Running code replacements on... `tests/test-helper.js`
+
+If you go back to the modifications done in `tests/index.html`, you can see the new script imports ` { start } from './test-helper`, registers the test files then call `start`. The `start` function is the one the codemod creates at the present step.
+
+Instead of loading the tests then calling `start` from `ember-qunit` directly, we rather export a `start` function that can be called later.
```

Comment:
```suggestion
Instead of loading the tests then calling `start` from `ember-qunit` directly in the module scope of the file, we instead export a `start` function that can be called later.
```